lua/RenderEvents.cpp

```diff
@@ -51,6 +51,17 @@ HOOK_METHOD(CommandGui, RenderPlayerShip, (Point &shipCenter, float jumpScale) -
     Global::GetInstance()->getLuaContext()->getLibScript()->call_on_render_event_post_callbacks(RenderEvents::LAYER_PLAYER, std::abs(idx), 0);
 }
 
+HOOK_METHOD(Ship, OnRenderSparks, () -> void)
+{
+    LOG_HOOK("HOOK_METHOD_PRIORITY -> Ship::OnRenderSparks -> Begin (RenderEvents.cpp)\n")
+    auto context = Global::GetInstance()->getLuaContext();
+    lua_pushinteger(context->GetLua(), this->iShipId);
```

Comment:
Why not push "this" itself here? Since we already have the pointer.

If "Ship" isn't exposed we can expose it (and at least its iShipId property).

Reply:
Done, `Ship` is exposed already